federated_aws_cli/cache.py

```diff
@@ -167,6 +167,45 @@ def write_id_token(issuer, client_id, token):
         logger.debug("Unable to write id token to: {}".format(path))
 
 
+@_requires_safe_cache_dir
+def read_sts_credentials(role_arn):
+    # Create a sha256 of the endpoint url, so fix length and remove weird chars
+    path = os.path.join(cache_dir, "stscreds_" + sha256(role_arn.encode("utf-8")).hexdigest())
+
+    if not os.path.exists(path) or _readable_by_others(path):
+        return None
+
+    try:
+        with open(path, "r") as f:
+            sts = json.load(f)
+
+            exp = datetime.datetime.strptime(sts["Expiration"], '%Y-%m-%dT%H:%M:%SZ').timestamp()
+            if exp - time.time() > CLOCK_SKEW_ALLOWANCE:
+                logger.debug("Using STS credentials at: {}, expiring in: {}".format(path, exp - time.time()))
+                return sts
+            else:
+                logger.debug(
+                    "Cached STS credentials have expired.".format(path))
+                return None
+    except (IOError, PermissionError):
+        logger.debug("Unable to read STS credentials from: {}".format(path))
+        return None
+
+
+@_requires_safe_cache_dir
+def write_sts_credentials(role_arn, sts_creds):
+    # Create a sha256 of the endpoint url, so fix length and remove weird chars
+    path = os.path.join(cache_dir, "stscreds_" + sha256(role_arn.encode("utf-8")).hexdigest())
+
+    try:
+        with _safe_write(path) as f:
+            json.dump(sts_creds, f, indent=2)
+
+            logger.debug("Successfully wrote STS credentials to: {}".format(path))
+    except (IOError, PermissionError):
```

Comment:
`PermissionError` isn't present prior to Python 3.3. Leverage this fix I applied earlier : 35395bc96b16a1e31ed92d418431650a4641a096